compiler/src/main/java/com/dylibso/chicory/compiler/internal/WasmAnalyzer.java

```diff
@@ -225,6 +265,58 @@ public List<CompilerInstruction> analyze(int funcId) {
                     result.add(new CompilerInstruction(CompilerOpCode.SWITCH, operands));
                     result.addAll(unwinds);
                     break;
+                case TRY_TABLE:
+                    {
+                        stack.enterScope(ins.scope(), blockType(ins));
```

Comment:
Those blocks are doing ways too much, and it's not your fault @chirino you have just been working around the current design.

Let me follow up with something that might enable cleaning this mess around.